### PR TITLE
Prevent superfluous restyle.

### LINKED ISSUE #79: Prevent extra re-styles by checking the current value first
See https://bugzilla.mozilla.org/show_bug.cgi?id=1199327#c22

### PR DESCRIPTION
**Pull request by etiennesegonzac**
Fixes #79

### CODE REVIEW

File: examples/async-chunks/app.js [hunk 1]
```diff
@@ -54,8 +54,14 @@
...
-      title.firstChild.data = record ? record.title : '';
-      body.firstChild.data = record ? record.body : '';
+      var titleContent = record ? record.title : '';
+      var bodyContent = record ? record.body : '';
+      if (title.firstChild.data !== titleContent) {
+        title.firstChild.data = titleContent;
+      }
+      if (body.firstChild.data !== bodyContent) {
+        body.firstChild.data = bodyContent;
+      }
```
Comment (etiennesegonzac) [inline examples/async-chunks/app.js:64]: looks like this is totally worth doing in the component too!

File: fast-list.js [hunk 2]
```diff
@@ -682,15 +682,20 @@ function placeItem(item, index, section, geometry, source, reload) {
 
 function resetTransform(item) {
   var position = item.dataset.position;
-  item.style.webkitTransform =
-    item.style.transform = 'translate3d(0, ' + position + 'px, 0)';
   item.dataset.tweakDelta = '';
+
+  var transform = item.style.transform || item.style.webkitTransform;
+  var newTransform = 'translate3d(0px, ' + position + 'px, 0px)';
+  if (transform === newTransform) return;
```
Comment (etiennesegonzac) [inline fast-list.js:689]: any idea how to unit-test this @wilsonpage ?

Comment (wilsonpage) [inline fast-list.js:689]: I'm afraid not :(

**Comment by wilsonpage** [comment]
r+!

Do we still need a similar patch in gaia-fast-list-.js, even after the poplar change?

**Comment by etiennesegonzac** [comment]
Nope, just need to make sure to use the latest poplar :)

**Pull request merged**